chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -194,9 +195,9 @@ func (b *BitcoindNotifier) startNotifier() error {
 	)
 
 	b.bestBlock = chainntnfs.BlockEpoch{
-		Height:      currentHeight,
-		Hash:        currentHash,
-		BlockHeader: blockHeader,
+		Height: currentHeight,
+		Hash:   currentHash,
+		Block:  block,
```

Comment:
ie, what is the big downside to just calling `GetBlock` when we actually want the full block?

Reply:
ah ok - after seeing `(n *NeutrinoNotifier) handleBlockConnected` it looks like this is not a problem since neutrino will in any case only dispatch blocks to "handle" that we are interested in anyways?

Reply:
Yeah, I [commented](https://github.com/lightningnetwork/lnd/pull/8893#discussion_r1689632775) on this as well. But I think we only download the block of the **start** block when starting the notifier, not each individual one in a range. But we should definitely test that somehow...

Reply:
> does this mean that we fetch every single block and not just those that have txs in that we interested in?

I just checked the code, for full nodes we already do this anyway, but we need to optimize to make sure we only fetch the block once - think we fetch it in multiple places atm.

I think this approach no longer works, you are right `handleBlockConnected` will fetch interested blocks, but we do `HandleMissedBlocks` in `connectFilteredBlock`, so blocks will be fetched. I need to find another way to do this job🤦🏻.